wasi/types.go

```diff
@@ -0,0 +1,366 @@
+	EPROTONOSUPPORT
+	// EPROTOTYPE Protocol wrong type for socket.
+	EPROTOTYPE
+	// ERANGE Result too large.
+	ERANGE
+	// EROFS Read-only file system.
+	EROFS
+	// ESPIPE Invalid seek.
+	ESPIPE
+	// ESRCH No such process.
+	ESRCH
+	// ESTALE Reserved.
+	ESTALE
+	// ETIMEDOUT Connection timed out.
+	ETIMEDOUT
+	// ETXTBSY Text file busy.
+	ETXTBSY
+	// EXDEV Cross-device link.
+	EXDEV
+	// ENOTCAPABLE Extension: Capabilities insufficient.
+	ENOTCAPABLE
+)
+
+var errnoToString = [...]string{
+	ESUCCESS:        "ESUCCESS",
+	E2BIG:           "E2BIG",
+	EACCES:          "EACCES",
+	EADDRINUSE:      "EADDRINUSE",
+	EADDRNOTAVAIL:   "EADDRNOTAVAIL",
+	EAFNOSUPPORT:    "EAFNOSUPPORT",
+	EAGAIN:          "EAGAIN",
+	EALREADY:        "EALREADY",
+	EBADF:           "EBADF",
+	EBADMSG:         "EBADMSG",
+	EBUSY:           "EBUSY",
+	ECANCELED:       "ECANCELED",
+	ECHILD:          "ECHILD",
+	ECONNABORTED:    "ECONNABORTED",
+	ECONNREFUSED:    "ECONNREFUSED",
+	ECONNRESET:      "ECONNRESET",
+	EDEADLK:         "EDEADLK",
+	EDESTADDRREQ:    "EDESTADDRREQ",
+	EDOM:            "EDOM",
+	EDQUOT:          "EDQUOT",
+	EEXIST:          "EEXIST",
+	EFAULT:          "EFAULT",
+	EFBIG:           "EFBIG",
+	EHOSTUNREACH:    "EHOSTUNREACH",
+	EIDRM:           "EIDRM",
+	EILSEQ:          "EILSEQ",
+	EINPROGRESS:     "EINPROGRESS",
+	EINTR:           "EINTR",
+	EINVAL:          "EINVAL",
+	EIO:             "EIO",
+	EISCONN:         "EISCONN",
+	EISDIR:          "EISDIR",
+	ELOOP:           "ELOOP",
+	EMFILE:          "EMFILE",
+	EMLINK:          "EMLINK",
+	EMSGSIZE:        "EMSGSIZE",
+	EMULTIHOP:       "EMULTIHOP",
+	ENAMETOOLONG:    "ENAMETOOLONG",
+	ENETDOWN:        "ENETDOWN",
+	ENETRESET:       "ENETRESET",
+	ENETUNREACH:     "ENETUNREACH",
+	ENFILE:          "ENFILE",
+	ENOBUFS:         "ENOBUFS",
+	ENODEV:          "ENODEV",
+	ENOENT:          "ENOENT",
+	ENOEXEC:         "ENOEXEC",
+	ENOLCK:          "ENOLCK",
+	ENOLINK:         "ENOLINK",
+	ENOMEM:          "ENOMEM",
+	ENOMSG:          "ENOMSG",
+	ENOPROTOOPT:     "ENOPROTOOPT",
+	ENOSPC:          "ENOSPC",
+	ENOSYS:          "ENOSYS",
+	ENOTCONN:        "ENOTCONN",
+	ENOTDIR:         "ENOTDIR",
+	ENOTEMPTY:       "ENOTEMPTY",
+	ENOTRECOVERABLE: "ENOTRECOVERABLE",
+	ENOTSOCK:        "ENOTSOCK",
+	ENOTSUP:         "ENOTSUP",
+	ENOTTY:          "ENOTTY",
+	ENXIO:           "ENXIO",
+	EOVERFLOW:       "EOVERFLOW",
+	EOWNERDEAD:      "EOWNERDEAD",
+	EPERM:           "EPERM",
+	EPIPE:           "EPIPE",
+	EPROTO:          "EPROTO",
+	EPROTONOSUPPORT: "EPROTONOSUPPORT",
+	EPROTOTYPE:      "EPROTOTYPE",
+	ERANGE:          "ERANGE",
+	EROFS:           "EROFS",
+	ESPIPE:          "ESPIPE",
+	ESRCH:           "ESRCH",
+	ESTALE:          "ESTALE",
+	ETIMEDOUT:       "ETIMEDOUT",
+	ETXTBSY:         "ETXTBSY",
+	EXDEV:           "EXDEV",
+	ENOTCAPABLE:     "ENOTCAPABLE",
+}
+
+const (
+	// FunctionArgsGet reads command-line argument data.
+	//
+	// In WebAssembly 1.0 (MVP) Text format, this signature is:
```

Comment:
Just curious. Does that help to have this signature comment in the doc of API interface in `wasi.go` side too?

Reply:
I'll change the link so that we don't have to duplicate it, but it is more obvious where to go